### PR TITLE
feat(create-mud): use pnpm in templates, move to root so they can be installed/run

### PR DESCRIPTION
**Pull request by holic**
Some notes:
- Initially wanted to put templates in the same workspace as the rest of the packages (i.e. add `templates/*` to root `pnpm-workspace.yaml`), but I felt like the hoisting of certain dependencies made it a little hard to ensure these templates would run the same way once "cloned" with `create-mud`. I also wasn't sure about a "workspace in a workspace" (each template is also a pnpm workspace).
- I started by checking in `pnpm-lock.yaml` for each template then `.gitignore`'d it for a smaller diff. I think we might be better off leaving this out, so that dependencies can bump patch versions for security needs. CI should catch any changes for us. But curious if you feel strongly in another direction.
- Since each template is a monorepo, and pnpm is (IMO) the best package manager for a monorepo, I think it makes sense to be opinionated and just use pnpm for the templates, rather than trying to create some normalizing layer between package managers and how they define/manage monorepos/workspaces.

### CODE REVIEW

File: packages/create-mud/package.json [hunk 1]
```diff
@@ -6,24 +6,22 @@
...
+    "build": "tsup src/cli.ts --minify && ./scripts/copy-templates.sh",
+    "clean": "rimraf dist",
     "dev": "tsup src/cli.ts --watch",
     "prepublishOnly": "npm run clean && npm run build",
     "test": "echo \"TODO\"",
     "release": "npm publish --access=public"
   },
   "dependencies": {
-    "create-create-app": "git+https://github.com/holic/create-create-app#e5c8bb449033418ad8518fc73f72c1a07eab2577"
+    "create-create-app": "git+https://github.com/holic/create-create-app#74376c59b48a04aabbe94d9cacfe9cb1cecccd63"
```
Comment (holic) [inline packages/create-mud/package.json:21]: this version bump improves package manager detection

File: packages/network/src/workers/Recover.worker.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Message } from "@latticexyz/services/protobuf/ts/ecs-relay/ecs-relay";
-import { expose } from "threads";
+import { expose } from "threads/worker";
```
Comment (holic) [inline packages/network/src/workers/Recover.worker.ts:2]: gpt-4 figured this out after a couple rounds of pasting in build errors and trying out its suggestions 🤯

File: .gitignore [hunk 3]
```diff
@@ -11,3 +11,7 @@ lerna-debug.log
 .retype
 .env
 yarn-error.log
+
+# We don't projects created from templates to ignore their lockfiles, but we don't
```
Comment (alvrs) [inline .gitignore:15]:
```suggestion
# We don't want projects created from templates to ignore their lockfiles, but we don't
```

File: packages/create-mud/package.json [hunk 4]
```diff
@@ -6,24 +6,24 @@
...
-    "dist",
-    "templates"
+    "dist"
   ],
   "scripts": {
     "prepare": "pnpm build",
-    "build": "tsup src/cli.ts --minify",
-    "clean": "shx rm -rf lib",
+    "build": "tsup src/cli.ts --minify && ./scripts/copy-templates.sh",
+    "clean": "rimraf dist",
```
Comment (alvrs) [inline packages/create-mud/package.json:14]: do we need to add rimraf as a dev dependency in this package?

File: packages/network/src/workers/Recover.worker.ts (hunk 2 above)
Comment (alvrs) [reply packages/network/src/workers/Recover.worker.ts:2]: 🤯

File: packages/create-mud/src/cli.ts [hunk 5]
```diff
@@ -16,7 +16,8 @@ create("create-mud", {
   promptForEmail: false,
   promptForLicense: false,
   promptForTemplate: true,
-  caveat: ({ answers }) => `Done! Play in the MUD with \`cd ${answers.name}\` and \`yarn dev\``,
+  caveat: ({ answers, packageManager }) =>
+    `Done! Play in the MUD with \`cd ${answers.name}\` and \`${packageManager} run dev\``,
```
Comment (alvrs) [inline packages/create-mud/src/cli.ts:20]: We still require folks to use pnpm in the templates right? Should we add a note about that here?

**Review by alvrs — APPROVED** [review]
This is great!

**Comment by roninjin10** [comment]
> Initially wanted to put templates in the same workspace as the rest of the packages (i.e. add templates/* to root pnpm-workspace.yaml), but I felt like the hoisting of certain dependencies made it a little hard to ensure these templates would run the same way once "cloned" with create-mud. I also wasn't sure about a "workspace in a workspace" (each template is also a pnpm workspace).

One of the big benifits of pnpm is this isn't much of a concern.  The store can be in node modules, somewhere globally on your system ect. and it will always work the same.  So I wouldn't worry too much

File: packages/create-mud/package.json [hunk 6]
```diff
@@ -6,24 +6,24 @@
...
     "dev": "tsup src/cli.ts --watch",
     "prepublishOnly": "npm run clean && npm run build",
-    "test": "echo \"TODO\"",
+    "test": "pnpm test:minimal && pnpm test:react",
+    "test:minimal": "(dist/cli.js test-project --template minimal || true) && rimraf test-project",
+    "test:react": "(dist/cli.js test-project --template react || true) && rimraf test-project",
```
Comment (alvrs) [inline packages/create-mud/package.json:19]: would this let the CI fail if creating the template fails or would it still pass bc of the `|| true`?

Comment (holic) [reply packages/create-mud/package.json:19]: good catch!

File: packages/create-mud/src/cli.ts (hunk 5 above)
Comment (holic) [reply packages/create-mud/src/cli.ts:20]: probably, I think I can also force the package manager here too

**Comment by holic** [comment]
> One of the big benifits of pnpm is this isn't much of a concern. The store can be in node modules, somewhere globally on your system ect. and it will always work the same. So I wouldn't worry too much

are you suggesting we _should_ get this to live within the same workspace? how does pnpm handle workspaces-within-workspaces? how do we protect against implicit dependencies (a template workspace not accidentally using a root workspace dep)

**Review by alvrs — APPROVED** [review]

**Pull request merged**